packages/blue-api-sdk/src/cache.ts

```diff
@@ -16,9 +26,86 @@ export const readMaybeBigInt = (value: string | number | undefined | null) => {
   return BigInt(value);
 };
 
+export const mergeArrayByField = <
+  T extends { __typename?: string },
+  Path extends DottedKeys<T> = DottedKeys<T>,
+>(
+  ...paths: Path[]
+) => {
+  const splittedPaths = paths.map((path) => path.split("."));
+
+  return (
+    existing: T[] | null | undefined,
+    incoming: T[],
+    {
+      readField,
+      mergeObjects,
+    }: {
+      readField: ReadFieldFunction;
+      mergeObjects: MergeObjectsFunction;
+    },
+  ) => {
+    const _get = (
+      // biome-ignore lint/suspicious/noExplicitAny: recursion breaks type
+      data: any,
+      path: string[],
+    ): PropertyKey | null | undefined => {
+      if (path.length === 0) return data;
+
+      const [key, ...rest] = path;
+
+      return _get(readField(key!, data), rest);
+    };
+
+    const getId = (
```

Comment:
This function deserves a comment since it took me a bit to really understand what it does

Reply:
esp. since it doesn't necessarily returns you an id but the value stored at the first one of `paths` at which a value is stored.

Reply:
_This function iterates over a list of possible `paths` and retrieves the first value found in `data` at any of these paths, or `undefined` if none is found._

Reply:
a better naming would be `getFirstValueAtPath` or sth like that imo